apps/ledger-live-desktop/src/renderer/drawers/DataSelector/SelectAccountAndCurrencyDrawer.tsx

```diff
@@ -48,32 +51,64 @@ const HeaderContainer = styled.div`
   align-items: center;
   justify-content: center;
 `;
+
+const ToggleContainer = styled.div`
+  padding: 0px 40px 16px 40px;
+  flex: 0 1 auto;
+`;
+
+const SearchInputContainer = styled.div`
+  padding: 0px 40px 16px 40px;
+  flex: 0 1 auto;
+`;
+
+const optionsToggle = [
+  {
+    value: "allCurrency",
+    label: <Trans i18nKey="drawers.selectCurrency.allCurrency" />,
+  },
+  {
+    value: "account",
+    label: <Trans i18nKey="drawers.selectCurrency.account" />,
+  },
+];
+
 export type SelectAccountAndCurrencyDrawerProps = {
   onClose?: () => void;
   currencies: CryptoOrTokenCurrency[];
   onAccountSelected: (account: AccountLike, parentAccount?: Account) => void;
   accounts$?: Observable<WalletAPIAccount[]>;
+  showAccountFilter?: boolean;
 };
-const SearchInputContainer = styled.div`
-  padding: 0px 40px 16px 40px;
-  flex: 0 1 auto;
-`;
 
 function SelectAccountAndCurrencyDrawer(props: SelectAccountAndCurrencyDrawerProps) {
-  const { currencies, onAccountSelected, onClose, accounts$ } = props;
+  const { currencies, onAccountSelected, onClose, accounts$, showAccountFilter } = props;
   const { t } = useTranslation();
   const [searchValue, setSearchValue] = useState<string>("");
+  const [filterWithAccounts, setFilterWithAccounts] = useState<string>("allCurrency");
```

Comment:
you should type this no? instead of string but use a type or an enum for type-safety?